examples/webgl2_texture3d_compressed_partialupdate.html

```diff
@@ -0,0 +1,237 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js webgl2 - texture3d compressed partial update</title>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+		<link type="text/css" rel="stylesheet" href="main.css">
+		<style>
+			body {
+				color: #222;
+			}
+
+			a {
+				color: #2fa1d6;
+			}
+
+			p {
+				max-width: 600px;
+				margin-left: auto;
+				margin-right: auto;
+				padding: 0 2em;
+			}
+		</style>
+	</head>
+
+	<body>
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> webgl2 - texture3d compressed partial update<br />
+			<p>
+		</div>
+
+		<!-- Import maps polyfill -->
+		<!-- Remove this when import maps will be widely supported -->
+		<script async src="https://unpkg.com/es-module-shims@1.3.6/dist/es-module-shims.js"></script>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.module.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+
+			import { KTX2Loader } from 'three/addons/loaders/KTX2Loader.js';
+
+			let container;
+			let camera, scene, renderer;
+
+			init();
+			animate();
+
+			function init() {
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 0.25, 100 );
+				camera.position.set( 0, 0, 5 );
+
+				scene = new THREE.Scene();
+
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				container.appendChild( renderer.domElement );
+
+
+				scene.background = new THREE.Color( 0xbbbbbb );
+
+				const ktx2Loader = new KTX2Loader();
+				ktx2Loader.setTranscoderPath( 'jsm/libs/basis/' );
+				ktx2Loader.detectSupport( renderer );
+
+				function obc( material ) {
+
+					material.onBeforeCompile = function ( shader ) {
+
+						shader.vertexShader = `
+						varying vec3 vWorldPosition;
+						attribute float textureId;
+						flat varying int vTextureId;
+						${shader.vertexShader}`;
+
+						shader.vertexShader = shader.vertexShader.replace(
+							'#include <worldpos_vertex>',
+							`
+							#include <worldpos_vertex>
+							vTextureId = int(gl_InstanceID);
+							vWorldPosition = position.xyz;`
+						);
+
+						shader.fragmentShader = shader.fragmentShader.replace(
+							'#include <map_pars_fragment>',
+							`
+							varying vec3 vWorldPosition;
+							flat varying int vTextureId;
+							precision highp sampler2DArray;
+							uniform sampler2DArray map;`
+						);
+
+						shader.fragmentShader = shader.fragmentShader.replace(
+							'#include <map_fragment>',
+							`
+								diffuseColor = texture( map, vec3( vMapUv, vTextureId ) );
+
+							`
+						);
+
+					};
+
+				}
+
```

Comment:
Do we really need a new example for this?

In any event,  would like to avoid modifying built-in materials whenever possible.

Reply:
Agreed 👍

Reply:
Should be good now!